### DISCUSSION TITLE
Email send improvement

### DISCUSSION THREAD

**Discussion by romanicak** [berrystephenw/OctoText#66] [Ideas]
...continue from:
https://github.com/berrystephenw/OctoText/discussions/62

I would not hesitate to donate for feature request like this:

User choose email or sms (combobox?)
Fields needed for email:
smtp server
smtp port (defines type of encryption/authentication used,-- 25, 465, 587 or whatever)
account username (=sender address, email from)
account pass
recipient address (= email to)

Also incremental print % need to put comma separated values. For me it would be good to put in ie. "50,90" , so I will know that half print is done and then in some short time the print will be finished. This helps time planning :-)

And I would also appreciate customizable texts for emails, but this is nice to have.

**Comment by berrystephenw** [comment]
I can look at CSV for print progress, but I was trying to avoid parsing input strings from the user. Getting this correct so it is easy to do seemed like a lot of work. I'll put it on the list for future enhancements.

Why do you want custom strings for messages? Is it for language?

**Reply by romanicak** [reply]
Yeah, I understand - input strings can be security nightmare :-) Maybe just simple regex alllowing only letters and numbers will be OK.

Not because of language, but because of make the text as short as possible. Mobile operators have something called "sms email" - You can send email to email address and they deliver it as sms to your phone. It is very fast, almost instant, no delay. Speed of delivery is essential for notifications :-)

But much more important is ability do deliver to address different from sender. Some smaller internet providers are blocking smtp servers other than their own, inside internal network. So use smtp server (and also account for it) needs to be separated from recipient mail address. Please will this be possible?

**Comment by berrystephenw** [comment]
I’m confused! The intent of octotext was to send text messages. This is the way it works now! I receive text messages from my outlook account, sent to my carrier’s SMS gateway everyday.
[Image.jpeg]

Stephen Berry
Hardware Engineering Manager
GSSI
40 Simon St
Nashua, NH 03060
________________________________

**Reply by romanicak** [reply]
You're lucky. Most of carrier’s SMS gateways are not usable for users direct like this. O2, vodafone, T-Mobile - none of them have public sms gateways, only in internal networks. (here in Europe, what I know).

I think it would be more user friendly, if user can use any known smtp server on which he has account created (= no need to create new email address just for sending) and also send notification to different email/smsemail at the same time.

**Reply by Bendermk6** [reply]
When i installed the plugin few weeks ago i had done maybe 15min research on google for the sms gateway for my provider. After that 15min it was clear to me, that here in germany sms-to-mail is not gonna work, because of my provider.

You can setup any mail smtp provider/server you want, that's what the plugin can do from the beginning.

**Reply by romanicak** [reply]
> You can setup any mail smtp provider/server you want, that's what the plugin can do from the beginning.

Yes, but You can't send mail to different recipient then the one used for smtp authentication. There's mischmash definitely, plugin is concatenating some fields and using them not the right way. All what I say is this could be feature request - or change request - to have an ability to separate login information of smtp server from destination email address.

**Reply by Bendermk6** [reply]
I'm sure you can use different sender and recipient email provider, i tested it with seznam.cz (sender) to gmail.com (recipient)

https://github.com/berrystephenw/OctoText/discussions/62#discussioncomment-630376

And yes, the user interface is not so much straight forward, but it was written for sending SMS first.
I guess email was just optional.
